build: enhance usage of libwhich path resolution to respect SONAME

vtjnash opened the pull request:
If the SONAME is available, it directly names the file we want to link to, so use that instead.

ararslan commented on base/Makefile:
```diff
@@ -130,6 +130,14 @@ resolve_path += && \
+else
+# try to use the SO_NAME (if the named file exists)
+resolve_path += && \
+	$1_=`objdump -p "$${$1}" | awk '/SONAME/ {print $$2}'` && \
```
I guess Awk should be added to the README section on required build tools?

staticfloat replied:
I actually wonder if `awk` is not required by POSIX?  It's builtin to `busybox`, which is my poor-mans litmus test to know whether or not it's available everywhere.  ;)

vtjnash replied:
fwiw, we already use it (to generate julia_version.h) – I checked before I used it here :)

staticfloat commented on deps/llvm.mk:
```diff
@@ -58,6 +58,12 @@ LLVM_CPPFLAGS :=
+LLVM_CPPFLAGS :=
+LLVM_LDFLAGS :=
+LLVM_CMAKE :=
+
```
This looks to be an unrelated change

staticfloat replied:
@vtjnash is this actually important in this PR?

vtjnash replied:
Not really ... and I just realized this is already in the Makefile right above it. Apparently this was just a rebase mistake – whereas I thought it was something I had done locally.

vtjnash committed:
build: enhance usage of libwhich path resolution to respect SONAME

If the SONAME is available, it directly names the file we want to link to, so use that instead.

vtjnash commented:
@staticfloat updated
